chunks 1,8: Do not set date to UNDATED (to be pushed upstream).
chunk 2: Tweak page headers (local change).
chunk 3: Put architecture name in parentheses (to be pushed upstream).
chunk 4: Default value for the .Os macro (local change).
chunks 5-7: No parentheses if the section is empty (to be pushed upstream).
chunk 9: Do not use an ugly double dash in the NAME line (local change).
chunk 10: Set the base paragraph indentation to 5n (upstream commit 5d2e49f8).

Index: tmac/mdoc/doc-common
--- tmac/mdoc/doc-common.orig
+++ tmac/mdoc/doc-common
@@ -278,7 +278,7 @@
 .\" NS doc-date-string global string
 .\" NS   the manual page date as set by 'Dd'
 .
-.ds doc-date-string UNDATED\"
+.ds doc-date-string \" empty
 .
 .
 .\" NS Dd user macro (not parsed, not callable)
@@ -335,7 +335,7 @@
 .ds doc-volume-ds-1 General Commands Manual
 .ds doc-volume-ds-2 System Calls Manual
 .ds doc-volume-ds-3 Library Functions Manual
-.ds doc-volume-ds-4 Kernel Interfaces Manual
+.ds doc-volume-ds-4 Device Drivers Manual
 .ds doc-volume-ds-5 File Formats Manual
 .ds doc-volume-ds-6 Games Manual
 .ds doc-volume-ds-7 Miscellaneous Information Manual
@@ -455,12 +455,11 @@
 .    ds doc-section \$2
 .    ie \B'\$2' \{\
 .      if ((\$2 >= 1) & (\$2 <= 9)) \{\
-.        ds doc-volume \" empty (not "LOCAL")
+.        ds doc-volume "\*[doc-volume-ds-\$2]
 .        if \A'\$3' \{\
 .          if d doc-volume-as-\$3 \
-.            as doc-volume "\*[doc-volume-as-\$3]
-.          \}
-.        as doc-volume " \*[doc-volume-ds-\$2]
+.            as doc-volume " (\*[doc-volume-as-\$3])
+.        \}
 .      \}
 .    \}
 .    el \{\
@@ -490,7 +489,7 @@
 .\" NS
 .\" NS override this in 'mdoc.local', if necessary
 .
-.ds doc-default-operating-system GNU\"
+.ds doc-default-operating-system OpenBSD\"
 .
 .
 .\" NS doc-operating-system global string
@@ -1005,7 +1004,8 @@
 .eo
 .de doc-header
 .  ds doc-xref \*[doc-page-topic-font]\*[doc-page-topic]\f[]\"
-.  as doc-xref \*[doc-page-section-font](\*[doc-section])\f[]\"
+.  if !"\*[doc-section]"" \
+.    as doc-xref \*[doc-page-section-font](\*[doc-section])\f[]\"
 .  ds doc-abbv \*[doc-page-topic]\"
 .  ev doc-env-dh
 .  doc-setup-page-layout
@@ -1016,7 +1016,8 @@
 .  if (\n[doc-reg-dh] + \n[doc-reg-dh1] + \n[doc-reg-dh] >= \n[.lt]) \{\
 .    while (\n[doc-reg-dh] + \n[doc-reg-dh1] + \n[doc-reg-dh] >= \n[.lt]) \{\
 .      ds doc-xref \*[doc-page-topic-font]\*[doc-abbv]\f[]\"
-.      as doc-xref \*[doc-page-section-font](\*[doc-section])\f[]\"
+.      if !"\*[doc-section]"" \
+.        as doc-xref \*[doc-page-section-font](\*[doc-section])\f[]\"
 .      length doc-abbv-len-prev \*[doc-abbv]
 .      substring doc-abbv 0 -2
 .      length doc-abbv-len \*[doc-abbv]
@@ -1061,7 +1062,8 @@
 .eo
 .de doc-footer
 .  ds doc-xref \*[doc-page-topic-font]\*[doc-page-topic]\f[]\"
-.  as doc-xref \*[doc-page-section-font](\*[doc-section])\f[]\"
+.  if !"\*[doc-section]"" \
+.    as doc-xref \*[doc-page-section-font](\*[doc-section])\f[]\"
 .  ds doc-page-id \n[%]
 .  if r X \{\
 .    if (\n[%] > \n[X]) \{\
@@ -1150,7 +1152,7 @@
 .  doc-break-page-with-new-number
 .
 .  \" Reset strings to reduce info leaks from one man page to the next.
-.  ds doc-date-string UNDATED\"
+.  ds doc-date-string \" empty
 .  ds doc-page-topic UNTITLED\"
 .  ds doc-volume LOCAL\"
 .  ds doc-section \" empty
@@ -1244,7 +1246,7 @@
 .
 .eo
 .de Nd
-.  nop \[em] \$*
+.  nop \[en] \$*
 ..
 .ec
 .
@@ -1482,7 +1484,7 @@
 .    if t \
 .      ss \n[doc-reg-Sh] \n[doc-reg-Sh1]
 .
-.    in \n[IN]u
+.    in \n[BP]u
 .    ns
 .
 .    doc-check-depth
